### PR TITLE
[docs] Introduce algolia search for doc site

### PR DESCRIPTION
**Pull request by anik120**
**Description of the change:**

This PR implements algolia search for https://sdk.operatorframework.io

**Motivation for the change:**

[Algolia](https://www.algolia.com/) is a powerful search API that is being used in a majority 
of open source documentation site. Besides creating a index for the site that is updated every 
hour, we get analytics like so: 

Screenshot from 2020-10-05 17-18-34

The [OLM doc site](https://olm.operatorframework.io) currently uses algolia search too. 

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Comment by hasbro17** [comment]
@asmacdo Any idea why this won't trigger a deploy preview for the docs? Are they configured to only run on changes in `website/content/*`.

**Comment by joelanford** [comment]
Repeating @hasbro17's question. Why isn't there a preview site for this change?

**Comment by anik120** [comment]
Same story as [this](https://github.com/operator-framework/operator-sdk/pull/4002#issuecomment-709516280),
opening #4048 kickstarted the preview for this PR.

**Comment by camilamacedo86** [comment]
We will need to rebase it with the master to get it merged. Could you please do that?

File: website/layouts/partials/hooks/body-end.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{{ with .Site.Params.algolia_docsearch }}
+<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.js"></script>
+<script type="text/javascript"> docsearch({
+apiKey: '265a4e0c97efa2f55b1f8a95a20335d9',
+indexName: 'sdk-operatorframework',
+inputSelector: '.td-search-input',
+debug: false // Set debug to true if you want to inspect the dropdown
+});
+</script>
+{{ end }}
```
Comment (camilamacedo86) [inline website/layouts/partials/hooks/body-end.html:10]:
```suggestion
{{ end }}

```

File: website/layouts/partials/hooks/head-end.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{{ with .Site.Params.algolia_docsearch }}
+<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/docsearch.js@2/dist/cdn/docsearch.min.css" />
+{{ end }}
```
Comment (camilamacedo86) [inline website/layouts/partials/hooks/head-end.html:3]:
```suggestion
{{ end }}

```

Comment (camilamacedo86) [reply website/layouts/partials/hooks/head-end.html:3]: Add the missing space at to end to fix it and to retrigger the CI after the migration to Travis ORG.

File: website/layouts/partials/hooks/body-end.html (hunk 1 above)
Comment (camilamacedo86) [reply website/layouts/partials/hooks/body-end.html:10]: add one empty caracter just to fix the CI because of the migration for the Travis.org

**Review by camilamacedo86 — APPROVED** [review]
It shows fine. 
The issue raised by @hasbro17 and @joelanford is solved
It is /lgtm for more than one person.

**Comment by camilamacedo86** [comment]
/hold

@anik120 I am testing the preview and before this change we return a list of result. Now, it is open the page of the first result found. Is it really the behaviour that we are looking for? 

c/c @asmacdo

**Comment by sujil02** [comment]
Closing is in favor of 4611

**Comment by sujil02** [comment]
Closing is in favor of PR 4611

**Pull request closed without merging**